pass_import.py

```diff
@@ -336,6 +337,212 @@ def clean(self, clean, convert):
+            raise FormatError()
+
+        entry['data'] = AppleKeychain._match_pair_to_dict(match, 'hex', 'txt')
+
+    @staticmethod
+    def _parse_field(line, entry):
+        match = re.search('^(?P<key>\w+):\s*(?P<hex>0x[a-zA-Z0-9]*)?\s*(?P<txt>.*)', line)
+        key = match.group('key')
+        if not match or not key:
+            raise FormatError()
+
+        data = AppleKeychain._match_pair_to_dict(match, 'hex', 'txt')
+        if 'txt' in data:
+            txt = data['txt']
+            if txt.startswith('"') and txt.endswith('"'):
+                data['txt'] = txt[1:-1]
+        entry[key] = data
+
+    @staticmethod
+    def _parse_note_plist(plist):
+        """Notes are stored in ASCII plist: extract the actual content."""
+        try:
+            tree = ElementTree.XML(plist)
+            dict = tree.find('dict')
+            return dict.find('string').text
+        except:
+            return None
+
+    @staticmethod
+    def _value_from_hextxt(data, default=None):
+        """Return a value from data with a hex-txt pair"""
+        value = default
+        if 'hex' in data:
+            decoded = AppleKeychain._decode_hex(data['hex'])
+            if decoded:
+                value = decoded
+            else:
+                if 'txt' in data:
+                    value = data['txt']
+                else:
+                    value = data['hex']
+        elif 'txt' in data:
+            value = data['txt']
+        return value
+
+    @staticmethod
+    def _convert_entry(entry):
+        """Returns an entry in pass format from an entry in keychain parsed format"""
+        human_keys = {
+                '0x00000007':'title',
+                'acct': 'login',
+                'atyp': 'authentication_type',
+                'cdat': 'creation_date',
+                'crtr': 'creator',
+                'desc': 'kind',
+                'icmt': 'alt_comment',
+                'mdat': 'modification_date',
+                'path': 'password_path',
+                'port': 'port',
+                'ptcl': 'protocol',
+                'sdmn': 'security_domain',
+                'srvr': 'server',
+                'svce': 'service',
+                'type': 'type'
+                }
+        passentry = {}
+        title = entry['attributes'].get('0x00000007', None)
+        for attr in entry['attributes']:
+            # Keychain duplicates 0x07 and svce by default
+            if attr == 'svce' and title == entry['attributes']['svce']:
+                continue
+            if attr in ['ptcl', 'srvr', 'path']:
+                continue
+
+            key = attr
+            if attr in human_keys:
+                key = human_keys[attr]
+
+            value = AppleKeychain._value_from_hextxt(entry['attributes'][attr])
+            if value:
+                passentry[key] = value
+
+        url = entry['attributes'].get('ptcl', {}).get('txt', '')
+        if url:
+            url = url.replace('htps', 'https')
```

Comment:
`url = url.replace('htps', 'https')` What is the purpose of this? There is no 'htps' in your test file

Reply:
It is the keyword used internally by Apple. See [protocol codes](https://gist.github.com/sangonz/601f4fd2f039d6ceb2198e2f9f4f01e0#protocol-codes). I've reproduced it in the test file and added more code substitutions.